### PR TITLE
fix installation issue that readiness-timeout flag is not honored.

### LINKED ISSUE #30221: istioctl readiness timeout flag ignored
**Bug description**
When using the --readiness-timeout flag with istioctl, the provided timeout value gets ignored, and it just uses the default 5m timeout.

[ ] Docs
[X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
istioctl should honor the provided readiness timeout

**Steps to reproduce the bug**
istioctl install -f custom.yaml -y --readiness-timeout 10m0s
^this is what I was using for a really long install, where there are 3 istio-egressgateway pods that take a long time to come up. It still times out after 5m0s and prints this message:
`✘ Egress gateways encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition`
you can try with other values.

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
istioctl: 
data plane version: 1.8.1 (1 proxies)

kubectl:
Client Version: v1.16.8
Server Version: v1.18.3+fa69cae

helm:
v3.1.2+gd878d4d

**How was Istio installed?**
Using the "Getting Started" doc
https://istio.io/latest/docs/setup/getting-started/

**Environment where the bug was observed (cloud vendor, OS, etc)**
IBM Cloud for provider, Mac OS 11.1 for client OS

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

### PR DESCRIPTION
**Pull request by morvencao**
resolve: https://github.com/istio/istio/issues/30221

[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: releasenotes/notes/30221.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+issue:
+  - 30221
+
+releaseNotes:
+- |
+  **Fixed** installation issue that readiness-timeout flag is not honored.
```
Comment (shamsher31) [inline releasenotes/notes/30221.yaml:9]:
```suggestion
  **Fixed** installation issue where `--readiness-timeout` flag is not honored.
```

Comment (morvencao) [reply releasenotes/notes/30221.yaml]: thanks @shamsher31 
done.

File: operator/pkg/helmreconciler/reconciler.go [hunk 2]
```diff
@@ -94,17 +94,19 @@ func NewHelmReconciler(client client.Client, restConfig *rest.Config, iop *value
 	if opts.ProgressLog == nil {
 		opts.ProgressLog = progress.NewLog()
 	}
-	if waitForResourcesTimeoutStr, found := os.LookupEnv("WAIT_FOR_RESOURCES_TIMEOUT"); found {
-		if waitForResourcesTimeout, err := time.ParseDuration(waitForResourcesTimeoutStr); err == nil {
-			opts.WaitTimeout = waitForResourcesTimeout
+	if int64(opts.WaitTimeout) == 0 {
```
Comment (shamsher31) [inline operator/pkg/helmreconciler/reconciler.go:97]: I don't understand this condition, why it needs to check for `WaitTimeout==0`

Comment (morvencao) [reply operator/pkg/helmreconciler/reconciler.go:97]: If the `WaitTimeout` is not zero value, then apply the default value.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**